File: p1732.java
class Solution {
    public int largestAltitude(int[] gain) {
        int[] altitude = new int[gain.length + 1];
        altitude[0] = 0;
        int temp = altitude[0];
        int greatest = altitude[0];

        for (int i = 0; i < gain.length; i++){
            temp += gain[i];
            altitude[i + 1] = temp;

            if (altitude[i + 1] > greatest){
                greatest = altitude[i + 1];
            }
        }

        return greatest;
    }
}

File: p1732.py
class Solution(object):
    def largestAltitude(self, gain):
        """
        :type gain: List[int]
        :rtype: int
        """

        altitude = []
        altitude.append(0)
        temp = altitude[0]
        greatest = altitude[0]

        for index, alt in enumerate(gain):
            temp += alt
            altitude.append(temp)

            if altitude[index + 1] > greatest:
                greatest = altitude[index + 1]

        return greatest